kinto/plugins/openid/__init__.py

```diff
@@ -0,0 +1,159 @@
+from jose import jwt
+from pyramid import authentication as base_auth
+from pyramid.interfaces import IAuthenticationPolicy
+from zope.interface import implementer
+
+import requests
+
+from kinto.core import utils
+from kinto.core.openapi import OpenAPI
+
+from kinto.core import logger
+
+
+def fetch_openid_config(issuer):
+    resp = requests.get(issuer.rstrip("/") + "/.well-known/openid-configuration")
+    return resp.json()
+
+
+@implementer(IAuthenticationPolicy)
+class OpenIDConnectPolicy(base_auth.CallbackAuthenticationPolicy):
+    def __init__(self, realm='Realm'):
+        self.realm = realm
+
+        self._openid_config = None
+        self._jwt_keys = None
+
+    def unauthenticated_userid(self, request):
+        """Return the userid or ``None`` if token could not be verified.
+        """
+        issuer = request.registry.settings["oidc.issuer_url"]
+        client_id = request.registry.settings["oidc.client_id"]
+        userid_field = request.registry.settings.get("oidc.userid_field", "sub")
+
+        authorization = request.headers.get('Authorization', '')
+        try:
+            authmeth, payload = authorization.split(' ', 1)
+        except ValueError:
+            return None
+
+        header_type = request.registry.settings.get("oidc.header_type", "bearer")
+        if authmeth.lower() != header_type:
+            return None
+
+        try:
+            # Bearer+OIDC id_token=jwt, access_token=bearer
```

Comment:
Remove support for ID_Token (which is used by the frontend), on the backend we want to use access/bearer token

Reply:
The thing is that we can have access tokens in JWT format. I haven't looked at those yet. 

Also, using id tokens saves a round-trip to the Identity Provider. With the inconvenient that their validity remains unquestioned until they expire.